### PR TITLE
(#215) Fix scripts, Makefiles to have test.sh run on other Linux dev boxes.

### PR DESCRIPTION
**Pull request by gapisback**
This commit does some clean-up of scripts and Makefiles so that the newly-added test.sh can be used by other dev engineers on diff Linux physical machines.

- Fix few Makefiles to allow re-specifying LOCAL_LIB, to point to, say, /usr/local/lib (for SSL libs, e.g.)
- Hacky support to allow use of OpenSSL V1.x versions

- Add NO_ENABLE_SEV=1 flag in certifier.mak, so that we can build
      shared library without #include'ing SEV_SNP-specific code. This
      seems to cause pytests which to "hang" as they end up calling
      SEV_SNP code-paths.

    Use this new flag when building shared-libraries in the  test-run_example-simple_app() test-case, in test.sh .
        This is used to execute pytests that need Certifier Service.

---
**NOTE**:  It’s not a complete fix, i.e., it does not address all the points Rado brought up. I just did enough clean-up to get test.sh running reasonably stably on Ye’s SGX box, w/o destabilizing CI jobs.

### CODE REVIEW

File: CI/scripts/test.sh [hunk 1]
```diff
@@ -231,8 +231,8 @@ function unit-test-certlib-utility-programs() {
     pushd utilities > /dev/null 2>&1
 
     # Build utilities
-    make -j2 -f cert_utility.mak
-    make -j2 -f policy_utilities.mak
+    make -j${NumMakeThreads} -f cert_utility.mak
+    make -j${NumMakeThreads} -f policy_utilities.mak
```
Comment (gapisback) [inline CI/scripts/test.sh:235]: Minor, slightly unrelated change. Fix straggling usages of `-j2` to specify # of threads as determined on host.

File: CI/scripts/test.sh [hunk 2]
```diff
@@ -424,7 +424,7 @@ function test-run_example-simple_app() {
 
     # Rebuild shared library that pytest needs
     pushd ../src > /dev/null 2>&1
-    make -f certifier.mak --always-make -j2 sharedlib
+    NO_ENABLE_SEV=1 make -f certifier.mak --always-make -j${NumMakeThreads} sharedlib
```
Comment (gapisback) [inline CI/scripts/test.sh:427]: The reason for this has been explained in [this note](https://github.com/vmware-research/certifier-framework-for-confidential-computing/issues/215#issuecomment-1783620655) added to the comments in the issue #215

File: src/certifier.mak [hunk 3]
```diff
@@ -1,7 +1,9 @@
 #    
 #    File: certifier.mak
 
-ENABLE_SEV=1
+ifndef NO_ENABLE_SEV
+    ENABLE_SEV=1
```
Comment (gapisback) [inline src/certifier.mak:5]: See [the reasons](https://github.com/vmware-research/certifier-framework-for-confidential-computing/issues/215#issuecomment-1783620655) explaining why this was introduced. 

This NO_ENABLE_SEV is OFF by default. It's only invoked currently by one `make` invocation in `test.sh`, so none of the other usages that build a shared lib will be affected.

File: tests/gen_client_server_certs_key_files.sh [hunk 4]
```diff
@@ -34,6 +34,12 @@ pushd "${TEST_DATA}" > /dev/null 2>&1
 # In more current OpenSSL v3.0 and later, use -noenc.
 # If your OpenSSL installation is older, change this to -nodes
 pvt_key_encr_arg="-noenc"
+set +e
+open_ssl_ver_is_1x=$(openssl version | grep -c "OpenSSL 1\.")
+set -e
+if [ "${open_ssl_ver_is_1x}" -eq 1 ]; then
+    pvt_key_encr_arg="-nodes"
```
Comment (gapisback) [inline tests/gen_client_server_certs_key_files.sh:41]: hacky-check to allow use of OpenSSL v1.x.

**Comment by gapisback** [comment]
@yelvmw , @rgerganov -- CI-jobs have passed with these changes. If you can take a quick look, I can try to merge this to /main.

**Review by rgerganov — APPROVED** [review]

**Review by yelvmw — APPROVED** [review]
